code/processes/tickerlogreplay.q

```diff
@@ -80,12 +81,11 @@ if[`.replay.usage in key .proc.params; -1 .proc.getusage[]; exit 0];
 // Check if some variables are null
 // some must be set
 .err.exitifnull each `.replay.schemafile`.replay.hdbdir, $[all null (tplogdir;tplogfile); `.replay.tplogfile; ()];
-
 if[basicmode and (messagechunks within (0;-1 + 0W));
  .err.ex[`replayinit; "if using basic mode, messagechunks must not be used (it should be set to 0W). basicmode will use .Q.hdpf to overwrite tables at the end of the replay";1]];
 if[not partitiontype in `date`month`year; .err.ex[`replayinit;"partitiontype must be one of `date`month`year";1]];
-
 if[messagechunks=0;.err.ex[`replayinit;"messagechunks value cannot be 0";2]];
+if[segmentedmode and (0<>firstmessage or 0W<>lastmessage);.err.ex[`replayinit;"firstmessage must be 0 and lastmessage must be 0W while in segmented mode"]]
```

Comment:
.err.ex needs a 3rd param I thnk (the exit code)
This condition is wrong due to left-of-right: (0<>firstmessage or 0W<>lastmessage)